### PR TITLE
Waylon/bin docs

### PR DESCRIPTION
**Pull request by 0xJepsen**
This PR
- Adds documentation to the binary crate
- Fixes the bind and init CLI args to use the arbiter template

### CODE REVIEW

File: bin/bind.rs [hunk 1]
```diff
@@ -1,25 +1,39 @@
-#![warn(missing_docs, unsafe_code)]
-
+#![warn(missing_docs)]
 use std::process::Command;
 
-pub(crate) fn bind_forge() -> std::io::Result<()> {
+use log::info;
```
Comment (Autoparallel) [inline bin/bind.rs:4]: You have to add an [`env_logger`](https://docs.rs/env_logger/latest/env_logger/) or else logs can't be displayed.

Logs might be overkill here, actually. A print to console is probably fine when it is simple and we always want a user to see it.

File: bin/main.rs [hunk 2]
```diff
@@ -12,64 +22,81 @@ use thiserror::Error;
 mod bind;
 mod init;
 
+/// Represents command-line arguments passed to the `Arbiter` tool.
 #[derive(Parser)]
 #[command(name = "Arbiter")]
 #[command(version = env!("CARGO_PKG_VERSION"))]
-#[command(about = "Data analysis tool for decentralized exchanges.", long_about = None)]
+#[command(about = "Ethereum Virtual Machine Logic Simulator", long_about = None)]
```
Comment (Autoparallel) [inline bin/main.rs:29]: This reminds me that we should go through and look for things like this and clean up our README and Github page in general.

File: Cargo.toml [hunk 3]
```diff
@@ -31,6 +31,10 @@ quote = "1.0.28"
 # Errors
 thiserror = "1.0.40"
 
+#logs
```
Comment (Autoparallel) [inline Cargo.toml:34]: It's a nitpick, but please format like everything else or things just look clunky

**Review by Autoparallel — CHANGES_REQUESTED** [review]
See the comments!

I also need to go run these commands myself to see what it's like first.

File: bin/main.rs (hunk 2 above)
Comment (0xJepsen) [reply bin/main.rs:29]: Yeah 100%

**Comment by 0xJepsen** [comment]
> See the comments!
> 
> I also need to go run these commands myself to see what it's like first.

Awesome! Yeah my recommendations are to run these in this order (and we should add these to the readme) I'll do that actually
- `arbiter init your-project-name`
- `cd your-project-name`
- `arbiter bind`
- `cargo run`

**Comment by Autoparallel** [comment]
> - cd your-project-name
> - arbiter bind

Should we add a possible input for bind to be a path to the `your-project-name/` directory and take `.` as a default?

**Comment by 0xJepsen** [comment]
> > * cd your-project-name
> > * arbiter bind
> 
> Should we add a possible input for bind to be a path to the `your-project-name/` directory and take `.` as a default?

I think that could be good in a later feature, but i think if we add that we should add it post release.

File: arbiter-core/src/environment.rs [hunk 4]
```diff
@@ -35,7 +35,7 @@ pub struct Environment {
     pub(crate) state: Arc<AtomicState>,
     pub(crate) evm: EVM<CacheDB<EmptyDB>>,
     pub(crate) socket: Socket,
-    pub seeded_poisson: SeededPoisson,
+    pub seeded_poisson: Option<SeededPoisson>,
```
Comment (Autoparallel) [inline arbiter-core/src/environment.rs:38]: Whoa wait, why are we doing this in this PR? Also this is a pretty hacky solution. It does work (and I did propose it lol) but can we get this out of here? It doesn't fit the intention of the PR.

File: Cargo.toml [hunk 5]
```diff
@@ -31,6 +31,10 @@ quote = "1.0.28"
 # Errors
 thiserror = "1.0.40"
 
+# Logs 
+log = "0.4.20"
+env_logger = "0.10.0"
+
```
Comment (Autoparallel) [inline Cargo.toml:37]: Are we using logs anywhere in the bin?

File: bin/init.rs [hunk 6]
```diff
@@ -1,149 +1,62 @@
...
+
+pub(crate) fn init_project(name: &str) -> io::Result<()> {
+    let status = Command::new("git")
+        .arg("clone")
+        .arg("https://github.com/primitivefinance/arbiter-template.git")
+        .arg(name)
+        .status()?;
+
+    if !status.success() {
+        error!("Failed to clone the repository.");
```
Comment (Autoparallel) [inline bin/init.rs:35]: Here, we are still logging.

**Review by Autoparallel — CHANGES_REQUESTED** [review]

**Review by Autoparallel — APPROVED** [review]

**Pull request merged**